src/server.py

```diff
@@ -85,36 +104,95 @@ def write_logs(self):
             responses = [json.loads(r) for r in self.responses]
             f.write(json.dumps(responses, indent=2))
 
+    def get_status(self):
+        training = False
+        if (self.train_proc is not None):
+            print("found training process, poll : {0}".format(self.train_proc.poll()))
+            if (self.train_proc.poll() is None):
+                training = True
+        models = glob.glob(os.path.join(self.model_dir, 'model*'))
+        return json.dumps({
+          "training": training,
+          "available_models": models
+        })
+
+    def auth(self, path):
+
+        if (self.token is None):
+            return True
+        else:
+            print("path : {0}".format(path))
+            parsed_path = urlparse.urlparse(path)
+            data = urlparse.parse_qs(parsed_path.query)
+            valid = ("token" in data and data["token"][0] == self.token)
+            return valid
+
+    def start_train_proc(self, data):
+        print("starting train")
+        if (self.train_proc is not None):
+            try:
+                self.train_proc.kill()
+            except:
+                pass
+        fname = 'tmp_training_data.json'
+        with open(fname, 'w') as f:
+            f.write(data)
+        cmd_str = "python -m rasa_nlu.train -c config.json -d {0}".format(fname)
```

Comment:
I'd rather use `multiprocessing` instead of `subprocess`. I think it will allow better control and error handling. Is there an advantage here in using `subprocess`?